onadata/apps/logger/data_migration/surveytree.py

```diff
@@ -63,9 +63,13 @@ def permanently_remove_field(self, field):
         """WARNING: It is not possible to revert this operation"""
         field.getparent().remove(field)
 
-    def modify_field(self, field, new_tag):
+    def change_field_tag(self, field, new_tag):
         field.tag = new_tag
 
+    def change_field_attrib(self, field, attrib, new_value):
+        if field.attrib.get(attrib) is not None:
+            field.attrib[attrib] = new_value
```

Comment:
why field.attrib.get does not raise exception when it's None? Shouldn't we get some sort of error here, since it might imply that migration tool has weird state?

Reply:
Why would .get() on dict raise exception? However, i wonder now whether this function should check if attrib is None, maybe instead is should just set it to value. Then we might rename it to `set_field_attrib` (i.e remove the if condition, leave the assignment)

Reply:
Sorry, I phrased that wrong, I meant we should probably cover the None case somehow, since right now it just does nothing when attrib does not exist. If it's ok to always set it, then `set_field_attrib` approach is fine

Reply:
Done